### PR TITLE
Classification methods: save/read parameters, improve negative value handling in log method

### PR DESCRIPTION
**Pull request by 3nids**

### CODE REVIEW

File: src/core/classification/qgsclassificationlogarithmic.cpp [hunk 1]
```diff
@@ -53,32 +53,52 @@ QIcon QgsClassificationLogarithmic::icon() const
...
+      minimum = positiveMinimum; // the minimum gets updated
+  }
+  else
+  {
+    positiveMinimum = minimum;
   }
 
   // get the min/max in log10 scale
-  double log_min = std::floor( std::log10( minimum ) );
+  double log_min = std::floor( std::log10( positiveMinimum ) );
```
Comment (nyalldawson) [inline src/core/classification/qgsclassificationlogarithmic.cpp]: What's with the python style naming?

**Review by nyalldawson — COMMENTED** [review]
LGTM!

**Review by nyalldawson — APPROVED** [review]

File: src/core/classification/qgsclassificationlogarithmic.cpp (hunk 1 above)
Comment (3nids) [reply src/core/classification/qgsclassificationlogarithmic.cpp]: what do you mean?

Comment (nyalldawson) [reply src/core/classification/qgsclassificationlogarithmic.cpp]: it should be logMin, not log_min... (I know it's petty, but if we're not super-strict on this then our api and code will become a 2.x style mess again....)

Comment (3nids) [reply src/core/classification/qgsclassificationlogarithmic.cpp]: oops, too much Python lately

**Pull request merged**